src/trio/_tests/type_tests/path.py

```diff
@@ -7,7 +7,7 @@
 from typing import IO, Any, BinaryIO, List, Tuple
 
 import trio
-from trio._path import _AsyncIOWrapper  # pyright: ignore[reportPrivateUsage]
+from trio._path import AsyncIOWrapper
```

Comment:
Why are we not doing `from trio._file_io import AsyncIOWrapper` here? And more interesting, why is this import not being marked as using a private import?

Reply:
No good reason, but I was trying to keep this PR somewhat self-contained rather than blarting changes anywhere I spotted something odd. 😅 

But you're quite right, this should be an import from a canonical location. Now it is. 💪 

Why `pyright` was okay with this I don't really know. I've had an argument with `pyright` recently and we're not talking to each other any more. 🙉